tensorboard/webapp/persistent_settings/_redux/persistent_settings_effects.ts

```diff
@@ -85,7 +92,11 @@ export class PersistentSettingsEffects implements OnInitEffects {
         // Buffers changes from all selectors and only emit when debounce period
         // is over.
         buffer(selectorsEmit$.pipe(debounceTime(DEBOUNCE_PERIOD_IN_MS))),
-        mergeMap((partialSettings) => {
+        withLatestFrom(this.store.select(getShouldPersistSettings)),
```

Comment:
Is it possible that we have `shouldPersistSettings=false` in L55 but still emit selector (and need L95 to guard again write settings into local storage) on loading shareable dashboard?
It seems to me `getSettings()` and `setSettings()` are guarded by the same selector but it's repeated in two places, which is a bit weird.

One possibility I can think of is, when a user loads a shareable dashboard and then updates some settings here (e.g. scalar smoothing), `selectorsEmit$` will be triggered (since that update fires navigating event), but still, it should be filtered by L55 as well?


Another maybe-related question is, the logic here is taking settings from local storage (L56), then do some check and load it to dashboard, but then write those (same) settings back to local storage (L106)? Why do we need to write back the settings we just take from local storage?